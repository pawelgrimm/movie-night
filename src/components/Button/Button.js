import React from "react";
import classNames from "classnames";

const Button = ({ type, children }) => {
  return (
    <button
      className={classNames("button", {
        "button--secondary": type === "secondary",
      })}
    >
      {children}
    </button>
  );
};

export default Button;
