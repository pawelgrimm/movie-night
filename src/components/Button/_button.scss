.button {
  background: $primary;
  border: none;
  border-radius: 3px;
  color: $on-primary;
  display: inline-block;
  font-size: $font-size-large;
  font-variant: all-small-caps;
  font-weight: 300;
  letter-spacing: 0.5px;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
  &:disabled {
    cursor: not-allowed;
    opacity: 0.5;
  }
  &:enabled {
    &:hover,
    &:focus {
      background: darken($primary, 3%);
      outline: none;
    }
    &:active {
      background: darken($primary, 5%);
      outline: none;
    }
  }
  & > svg {
    margin: auto;
  }
  @media (max-width: $desktop-breakpoint) {
    font-size: $font-size-medium;
    padding: $xs-size;
    & > svg {
      height: 16px;
      width: 16px;
    }
  }
}

.button--link {
  background: none;
}

.button--secondary {
  background: $secondary;
  &:disabled {
    opacity: 0.5;
    cursor: not-allowed;
  }
  &:enabled {
    &:hover,
    &:focus {
      background: darken($secondary, 3%);
      outline: none;
    }

    &:active {
      background: darken($secondary, 5%);
      outline: none;
    }
  }
}

.button-group {
  display: flex;
  flex-direction: row;
  margin: $xs-size 0;
  * {
    margin-left: 0;
    margin-right: $s-size;
    @media (max-width: $desktop-breakpoint) {
      margin-right: $xs-size;
    }
  }
  .button:last-child {
    margin-right: 0;
  }
  .button {
    flex-grow: 2;
  }
  .button--secondary {
    flex-grow: 1;
  }
  @media (max-width: $desktop-breakpoint) {
    margin: $sh-size;
  }
}
