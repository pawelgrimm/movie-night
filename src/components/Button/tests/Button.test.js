import { expect, jest } from "@jest/globals";
import React from "react";
import { shallow } from "enzyme";
import Button from "../Button";

test("should render buttons", () => {
  const button = shallow(<Button>Button Text</Button>);
  expect(button).toMatchSnapshot();
  const secondaryButton = shallow(
    <Button type="secondary">Button Text</Button>
  );
  expect(secondaryButton).toMatchSnapshot();
});

test("should have correct class", () => {
  const button = shallow(<Button>Button Text</Button>);
  expect(button.hasClass("button")).toBeTruthy();
});

test("secondary button should have correct classes", () => {
  const button = shallow(<Button type="secondary">Button Text</Button>);
  expect(button.hasClass("button")).toBeTruthy();
  expect(button.hasClass("button--secondary")).toBeTruthy();
});

test("should render children", () => {
  const button = shallow(
    <Button>
      <div id="inner">Some text</div>
    </Button>
  );
  expect(button.find("#inner").type()).toEqual("div");
});
