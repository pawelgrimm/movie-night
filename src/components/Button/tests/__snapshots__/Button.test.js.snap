// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render buttons 1`] = `
<button
  className="button"
>
  Button Text
</button>
`;

exports[`should render buttons 2`] = `
<button
  className="button button--secondary"
>
  Button Text
</button>
`;
